src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/DirectoryAttribute.cs

```diff
@@ -115,150 +103,140 @@ public object[] GetValues(Type valuesType)
                 for (int i = 0; i < count; i++)
                 {
                     if (List[i] is string)
+                    {
                         results[i] = (string)List[i];
+                    }
                     else if (List[i] is byte[])
+                    {
                         results[i] = s_encoder.GetString((byte[])List[i]);
+                    }
                     else
-                        throw new NotSupportedException(String.Format(CultureInfo.CurrentCulture, SR.DirectoryAttributeConversion));
+                    {
+                        throw new NotSupportedException(SR.DirectoryAttributeConversion);
+                    }
                 }
 
                 return results;
             }
-            else
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, SR.ValidDirectoryAttributeType), "valuesType");
+
+            throw new ArgumentException(SR.ValidDirectoryAttributeType, nameof(valuesType));
         }
 
         public object this[int index]
         {
             get
             {
-                if (!isSearchResult)
+                if (!_isSearchResult)
                 {
                     return List[index];
                 }
-                else
+
+                if (List[index] is byte[] temp)
                 {
-                    byte[] temp = List[index] as byte[];
-                    if (temp != null)
+                    try
                     {
-                        try
-                        {
-                            return s_utf8EncoderWithErrorDetection.GetString(temp);
-                        }
-                        catch (ArgumentException)
-                        {
-                            return List[index];
-                        }
+                        return s_utf8EncoderWithErrorDetection.GetString(temp);
                     }
-                    else
+                    catch (ArgumentException)
                     {
-                        // should not happen
                         return List[index];
                     }
                 }
+                
+                // This hould not happen.
```

Comment:
I'm fairly sure it can be reached in certain situations at this point, but I need to add some tests to make sure